browser/ui/views/brave_actions/brave_news_action_view.cc

```diff
@@ -117,18 +120,10 @@ void BraveNewsActionView::Update() {
     has_feeds = !tab_helper->GetAvailableFeeds().empty();
   }
 
-  auto background =
-      subscribed ? views::CreateRoundedRectBackground(
-                       selectedColor,
-                       ChromeLayoutProvider::Get()->GetCornerRadiusMetric(
-                           views::Emphasis::kMaximum, GetPreferredSize()))
-                 : nullptr;
-  auto image =
-      gfx::CreateVectorIcon(kBraveTodaySubscribeIcon, 16,
-                            color_utils::DeriveDefaultIconColor(
-                                subscribed ? SK_ColorWHITE : SK_ColorBLACK));
+  auto image = gfx::CreateVectorIcon(
+      kBraveTodaySubscribeIcon, 16,
+      color_utils::DeriveDefaultIconColor(GetIconColor(subscribed)));
   SetImage(ButtonState::STATE_NORMAL, image);
```

Comment:
Is there a reason why? It looks like `SetImage` just calls `SetImageModel` with an image model it creates via `ui::ImageModel::FromImageSkia(image)`, which is pretty much exactly what I'll do if I call `SetImageModel` from here.

Reply:
Oh, sorry. I forgot to describe why. SetImage() was supposed to be deprecated in a long term. 
https://bugs.chromium.org/p/chromium/issues/detail?id=1100034
https://source.chromium.org/chromium/chromium/src/+/main:ui/views/controls/button/label_button.h;l=50?q=label_button&ss=chromium%2Fchromium%2Fsrc

When we set an ImageModel to LabelButton and the ImageModel was constructed with VectorIcon and color_id, LabelButton will update icons automatically OnThemeChanged(). This would let child classes not have OnThemeChanged(). - but this is not our case here.

Reply:
Aaah right, makes sense. Well seeing as `SetImage` is going to be deprecated anyway I'll move over to `SetImageModel`. Cheers @sangwoo108